packages/core/src/tools/error.ts

```diff
@@ -110,3 +137,20 @@ export function createHandlingStack(): string {
 
   return formattedStack!
 }
+
+export function flattenErrorCauses(error: RawError, parentSource: ErrorSource): ErrorCause[] {
+  let count = 0
+  let currentError = error
+  const causes: Array<RawError | ErrorCause> = []
+  while (currentError?.cause && currentError.cause && count < 10) {
+    causes.push({
+      message: currentError.cause.message,
+      source: currentError.cause.source ?? parentSource,
+      type: currentError.cause?.type,
+      stack: currentError.cause?.stack,
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** you should "normalize" the stack by parsing then formating it again. Maybe refactor a bit to use the same logic as `formatUnknownError`?

Reply:
the schema asks for `stack` to be of type `string`. But ideally here we should have `stack` as type `StackTrace`. Should I update the schema?